### ISSUE TITLE
Improvement: Improve ranking algorithm 

### ISSUE THREAD

**Issue by anuraghazra** [anuraghazra/github-readme-stats#41]
The ranking algorithm works ok but i have seen some users where they have relatively low stats but still got A, A++ ranks.

So the ranking algorithm is currently biased and unbalanced, if someone can create a better one feel free to contribute :smile: 

file to look at: [calculateRank.js](https://github.com/anuraghazra/github-readme-stats/blob/master/src/calculateRank.js)

**Comment by garvit-joshi** [comment]
The Values of ```OFFSET's``` is not good Enough.
```
const COMMITS_OFFSET = 0.75;
const CONTRIBS_OFFSET = 1.65;
const ISSUES_OFFSET = 1;
const STARS_OFFSET = 1.25;
const PRS_OFFSET = 0.75;
const FOLLOWERS_OFFSET = 0.45;
```

**Comment by anuraghazra** [comment]
So what's you potential suggestions to improve @garvit-joshi? :thinking: maybe we could normalize the values

**Comment by garvit-joshi** [comment]
Yes @anuraghazra, Algorithm looks good, problem lies in values, I will notify you if I get a good solution.

**Comment by TheAshwanik** [comment]
#44

**Comment by vinayaksh42** [comment]
We can use the elo rating system. It will have a greater curve at the middle ranks and then eventually the top rank for a lesser number of people

**Comment by garvit-joshi** [comment]
#47 Updated Algorithm.

**Comment by anuraghazra** [comment]
> We can use the elo rating system. It will have a greater curve at the middle ranks and then eventually the top rank for a lesser number of people

@vinayaksh42 this seem reasonable, any ideas on how to implement it?

**Comment by vinayaksh42** [comment]
@anuraghazra we can setup a curve of rankings , for example if we have a range of ranking between 0-100. 
The top rank can be between 91-100.
The next rank can have a bigger range (76-90).
It will ensure that less number of people are in the top tier and more people in the middle and average tier.

For calculating the score of users we can change the offset according to which factor in the score is more common and which is less common.

**Comment by anuraghazra** [comment]
should be fixed in #49

**Comment by jrock2004** [comment]
Has this been released? I saw a user where their numbers were 0, but their rating shows A+ https://github-readme-stats.vercel.app/api?username=edward-rosado
